### PR TITLE
Remove hover & pointer on disabled ItemCheckbox

### PR DESCRIPTION
**Pull request by mcellier**
## Description

Impact both `ItemCheckbox` & `ItemRadio` components. Demo for `ItemCheckbox`:

Before:
before

After:
after

## What has been done

- Remove `cursor: pointer` when `disabled` is true
- Set `isClickable` prop to false when `disabled` is true to remove the background 

## Things to consider

Some snapshots will be updated when upgrading kairos

## How it was tested

With storybook

### CODE REVIEW

File: src/itemCheckbox/index.tsx [hunk 1]
```diff
@@ -11,6 +11,10 @@ const StyledItemCheckbox = styled(ItemCheckbox)`
     position: absolute;
     clip: rect(0, 0, 0, 0);
   }
+
+  &.kirk-item-checkbox--disabled {
+    cursor: default;
```
Comment (Tuxosaurus) [inline src/itemCheckbox/index.tsx:16]: Maybe remove the new class and use the prop instead (I received this same feedback when disabling icons):
```
& {
    cursor: ${props => (props.isDisabled ? 'default' : 'pointer')};
  }
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # Unreleased
 
 [...]
+- **[FIX]** Remove hover color & pointer when `disabled` is passed to `ItemCheckbox`
```
Comment (Tuxosaurus) [inline CHANGELOG.md:4]: I guess this should also be applied to `ItemRadio`

Comment (nickoola) [reply CHANGELOG.md:4]: Yep, at least create a ticket if you prefer doing it separatly @mcellier

File: src/itemCheckbox/index.tsx (hunk 1 above)
Comment (mcellier) [reply src/itemCheckbox/index.tsx]: Thanks, it's way cleaner that way

File: CHANGELOG.md (hunk 2 above)
Comment (mcellier) [reply CHANGELOG.md]: Done

**Comment by wakooka** [comment]
Would it be possible to do this on the Item directly only based on the `disabled` prop?
Probably need a bigger refacto.

**Comment by mcellier** [comment]
> Would it be possible to do this on the Item directly only based on the `disabled` prop?
> Probably need a bigger refacto.

All items aren't clickable or else. I don't want to touch the base `Item` too much, as it's a mess :/

**Pull request merged**